lib/query_components/text_query.rb

```diff
@@ -59,10 +77,17 @@ def should_conditions
       groups << field_boosts_synonyms unless search_params.disable_synonyms?
       groups << field_boosts_shingles
       groups << field_boosts_id_codes
+      dismax_groups(groups)
+
+      # groups.map { |queries|
+      #   dis_max_query(queries)
+      # }
+    end
 
-      groups.map { |queries|
+    def dismax_groups(groups)
+      groups.map do |queries|
```

Comment:
We try to use {} instead of `do` `end` when we care about the result of the iteration (not explicitly said in the styleguide, but a consequence of the "Never chain do...end" guideline)